src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -271,7 +272,7 @@ class MarathonSchedulerActor(
     withLockFor(Set(appId))(f)
 
   // there has to be a better way...
-  def driver: SchedulerDriver = MarathonSchedulerDriver.driver.get
+  def driver: SchedulerDriver = marathonSchedulerDriverHolder.driver.get
```

Comment:
The comment already suggests that. 
I did not understand completely, why we need this.